scipy/stats/_sensitivity_analysis.py

```diff
@@ -0,0 +1,717 @@
+from __future__ import annotations
+
+import inspect
+from dataclasses import dataclass
+from typing import (
+    Callable, Dict, List, Literal, Protocol, TYPE_CHECKING, Tuple
+)
+
+import numpy as np
+
+from scipy.stats._common import ConfidenceInterval
+from scipy.stats._qmc import check_random_state
+from scipy.stats._resampling import BootstrapResult
+from scipy.stats import qmc, bootstrap
+
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy._lib._util import DecimalNumber, IntNumber, SeedType
+
+
+__all__ = [
+    'sobol_indices'
+]
+
+
+def f_ishigami(x: npt.ArrayLike) -> np.ndarray:
```

Comment:
Correct me if I'm wrong, but wouldn't this error out if `TYPE_CHECKING` is set to `False` because `numpy.typing` would not be imported?

Or is this expected behaviour in the context of testing?

Reply:
My understanding is that at runtime types are not evaluated (thanks to `from __future__ import annotations`), but when mypy is invoked it is. Meaning in this case we need some extra imports. We could have the imports at runtime, I don't think in this case it would be an issue. This pattern is used usually to prevent deps cycles. Since we use this in some other places, I am following that for consistency.

Some reading here: https://mypy.readthedocs.io/en/stable/runtime_troubles.html